scripts/west_commands/packages.py

```diff
@@ -0,0 +1,145 @@
+                for a given package manager.
+                See 'west packages <manager> --help' for details.
+            """
+            ),
+        )
+
+        parser.add_argument(
+            "-p",
+            "--project",
+            action="append",
+            default=[],
+            dest="projects",
+            help="Project to run the 'packages' command for. "
+            "Use 'zephyr' if the 'packages' command should run for Zephyr itself. "
+            "Option can be passed multiple times. "
+            "If this option is not given, the 'packages' command will run for Zephyr "
+            "and all active projects.",
+        )
+
+        subparsers_gen = parser.add_subparsers(
+            metavar="<manager>",
+            dest="manager",
+            help="select a manager.",
+            required=True,
+        )
+
+        pip_parser = subparsers_gen.add_parser(
+            "pip",
+            help="manage pip packages",
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            epilog=textwrap.dedent(
+                """
+            Manage pip packages:
+
+                Run 'west packages pip' to print all pip packages or requirements
+                files needed by Zephyr and the active projects.
+
+                The output is compatible with the requirements file format itself.
+            """
+            ),
+        )
+
+        pip_parser.add_argument(
+            "--install",
+            action="store_true",
+            help="Install pip packages instead of listing them. "
+            "A single 'pip install' command is built and executed. "
+            "Additional pip arguments can be passed after a -- separator "
+            "from the original 'west packages pip --install' command. For example pass "
+            "'--dry-run' to not actually install anything, but print what would be.",
+        )
+
+        return parser
+
+    def do_run(self, args, unknown):
+        if len(unknown) > 0 and unknown[0] != "--":
+            self.die(
+                f'Unknown argument "{unknown[0]}"; '
+                'arguments for the manager should be passed after "--"'
+            )
+
+        if args.manager == "pip":
+            return self.do_run_pip(args, unknown[1:])
+
+        # Unreachable but print an error message if an implementation is missing.
+        self.die(f'Unsupported package manager: "{args.manager}"')
+
+    def do_run_pip(self, args, manager_args):
+        packages = []
+        requirements = []
+
+        if not args.projects or "zephyr" in args.projects:
+            requirements.append(ZEPHYR_BASE / "scripts/requirements.txt")
+
+        for module in zephyr_module.parse_modules(ZEPHYR_BASE, self.manifest):
+            module_name = module.meta.get("name", None)
+            if args.projects and module_name not in args.projects:
+                continue
+
+            pip = module.meta.get("package-managers", {}).get("pip")
```

Comment:
Short of using `git grep`, this line is really not obvious to me... can you please add a one-line comment with a clue where this data comes from?

I'm a bit confused by the terseness of the commit messages... is this effectively deprecating `scripts/requirements.txt` files and moving them to YAML?

Reply:
I've added more details in the commit message.
For `pip` it's not the intention to deprecate `scripts/requirements.txt` files, but refer to them form the module files.
I've added both the option to list individual packages or requirements files.

Reply:
I've added an example to the zephyr documentation, see https://builds.zephyrproject.io/zephyr/pr/80782/docs/develop/modules.html#package-manager-dependencies